src/components/catalog-search/catalog-search.controller.ts

```diff
@@ -60,6 +68,12 @@ export class CatalogSearchController implements angular.IController {
   private filterForKeywords(searchText: string) {
     let keywords = this.KeywordService.generateKeywords(searchText);
     let items = this.KeywordService.filterForKeywords(this.ctrl.catalogItems, ['name', 'tags'], keywords);
-    return _.take(items, 5);
+    let totalNumItems: number = _.size(items);
+    if (totalNumItems > this.maxResultsToShow) {
+      items.splice(this.maxResultsToShow, 0, {id: 'viewAll', name: searchText, totalNumResults: totalNumItems});
```

Comment:
Instead of using `splice` can you use `_.take(items, this.maxResultsToShow)` and then just append to the array? I think it is easier to follow that way.